src/main/webapp/app/pages/companyPage/CompanyPage.tsx

```diff
@@ -79,176 +81,170 @@ const LICENSE_STATUS_UPDATE_MESSAGES = {
-  @observable getCompanyStatus: PromiseStatus;
   @observable selectedLicenseStatus: LicenseStatus;
+
   @observable conflictingDomains: string[] = [];
 
   @observable showModal = false;
+
   @observable confirmModalText = '';
-  @observable formValues: any;
 
-  @observable companyTableLoading = true;
-  @observable companyUsers: UserDTO[] = [];
-  @observable companyUsersTokens: Token[] = [];
+  @observable formValues: any;
 
-  @observable userDropdownLoading = true;
-  @observable availableUsers: SelectOptionType[] = [];
   @observable selectedUsersOptions: SelectOptionType[] = [];
 
-  constructor(props: RouteComponentProps<MatchParams>) {
-    super(props);
-    this.getCompanyInfos();
-  }
+  @observable companyUserTokens: MobxPromiseUnionType<Token[]>;
 
-  @action
-  async getCompanyInfos() {
-    try {
-      this.getCompanyStatus = PromiseStatus.pending;
-      this.companyTableLoading = true;
-      this.userDropdownLoading = true;
-      this.company = await client.getCompanyUsingGET({
+  @observable companyUsers: MobxPromiseUnionType<UserDTO[]>;
+
+  @observable company = remoteData<CompanyDTO>({
+    invoke: async () => {
+      const company = await client.getCompanyUsingGET({
         id: parseInt(this.props.match.params.id),
       });
-      this.selectedLicenseStatus = this.company.licenseStatus as LicenseStatus;
-      this.getCompanyUsers();
+      this.selectedLicenseStatus = company.licenseStatus as LicenseStatus;
       this.verifyCompanyDomains();
-      this.getCompanyStatus = PromiseStatus.complete;
-    } catch (error) {
-      this.getCompanyStatus = PromiseStatus.error;
-      notifyError(error);
-    }
-  }
+      this.getCompanyInfo(company.id);
+      return company;
+    },
+  });
 
-  @action
-  getCompanyUsers() {
-    client
-      .getCompanyUsersUsingGET({
-        id: this.company.id,
-      })
-      .then(async companyUsers => {
-        this.companyUsers = companyUsers;
-        this.getDropdownUsers();
-        await this.getCompanyUsersTokens();
-        this.companyTableLoading = false;
-      })
-      .catch(error => notifyError(error));
-  }
-
-  @action
-  async getCompanyUsersTokens() {
-    this.companyUsersTokens = await client.getUsersTokensUsingPOST({
-      logins: this.companyUsers.map(user => user.login),
-    });
-  }
-
-  @action
-  async getDropdownUsers() {
-    try {
+  readonly nonCompanyUsers = remoteData<SelectOptionType[]>({
+    async invoke() {
       const allUsers = await client.getNonCompanyUserEmailsUsingGET({});
-      this.availableUsers = allUsers.map(email => ({
+      return allUsers.map(email => ({
         label: email,
         value: email,
       }));
-      this.userDropdownLoading = false;
-    } catch (error) {
-      notifyError(error);
+    },
+    default: [],
+  });
+
+  /* After the company is fetched, we will fetch the company users and the tokens */
+  @action.bound
+  getCompanyInfo(id: number) {
+    this.companyUsers = remoteData<UserDTO[]>({
```

Comment:
We should use `await` in the remoteData so when company is updated the companyUsers will be updated as well. This should be the case.